internal/controller/nimcache_controller.go

```diff
@@ -82,6 +84,7 @@ type NIMCacheReconciler struct {
 	log      logr.Logger
 	Platform platform.Platform
 	updater  conditions.Updater
+	recorder record.EventRecorder
```

Comment:
Where is this initialized? When we setup reconciler we need to initialize this with `recorder: mgr.GetEventRecorderFor("NIMCache"),

Reply:
Exactly. See for example https://github.com/NVIDIA/k8s-nim-operator/pull/155/files/cc926309989b6822baff489cd3cfdc91f4eb3697#diff-cf3e8a734e3ad6a908183ede7ac138abd176264cd8be708b7eedceed8b387ba0R193